src/main/java/com/google/research/bleth/simulator/Agent.java

```diff
@@ -0,0 +1,21 @@
+package com.google.research.bleth.simulator;
+
+/** Represent both observer and beacon. */
+public interface Agent {
+
+    /**
+     * @return the current agent's location on the real board.
+     */
+    Location getLocation();
+
+    /**
+     * Calculate the location the agent is moving to, based on its current location and its strategy.
+     * @return the location on board which the agent is moving to.
+     */
+    Location moveTo();
+
+    /**
+     * Move the agent to its next location and update the board accordingly.
+     */
+    void move();
+}
```

Comment:
And the implementation will be something like:
    public void move() {
        Location nextMove = moveTo();
        simulation.getBoard().moveAgent(realLocation, nextMove, this);
        realLocation = nextMove;
    }

Reply:
Looks good.